src/cp.ts

```diff
@@ -66,6 +66,11 @@ export class Cp {
                         }
                     },
                 )
+                .then((conn) => {
+                    conn.onclose = (event) => {
+                        resolve();
```

Comment:
There is a resolve above at line 63. I don't think that this resolve is required. In fact I'm not sure it's what we want, because this will resolve when the websocket is returned. We want to resolve when the copy completes.

Reply:
That callback only gets called if `status != null`: https://github.com/kubernetes-client/javascript/blob/8db906a63eff5bf296f3fac4572f83ea3db51283/src/exec.ts#L55

For copies (or maybe for all `exec` calls that use `stdin`), `status == null`, so the callback never gets called, so the promise never resolves. `await`ing calls to `cpToPod` hang forever.

Reply:
If the copy isn't complete when the websocket is closed, how can the caller ever now that it has completed?

Reply:
Ah, I missed that you are doing this on the 'close' event. My mistake. But we should probably make sure that we don't resolve the promise twice.

Reply:
OK, added a check for that in a way that shouldn't cause things to break if changes elsewhere cause this logic to change.